File: frontend/portfolioerp-invoices/src/models/invoice.ts
export interface InvoiceLine {
    id: number;
    productId: number;
    productCode: string;
    description: string;
    quantity: number;
    unitPrice: number;
    discountPercentage: number;
    discountAmount: number;
    netAmount: number;
    vatPercentage: number;
    vatAmount: number;
    totalAmount: number;
}

export interface Invoice {
    id: number;
    invoiceNumber: string;
    salesOrderId: number;
    customerId: number;
    customerName: string;
    invoiceDateUtc: string;
    status: number;
    subtotal: number;
    discountAmount: number;
    taxAmount: number;
    totalAmount: number;
    lines: InvoiceLine[];
}

File: backend/PortfolioERP.InvoiceService/Domain/Invoice.cs
namespace PortfolioERP.InvoiceService.Domain;

public class Invoice
{
    public int Id { get; set; }

    public string InvoiceNumber { get; set; } = null!;

    public int SalesOrderId { get; set; }

    public int CustomerId { get; set; }

    public string CustomerName { get; set; } = null!;

    public DateTime InvoiceDateUtc { get; set; }

    public InvoiceStatus Status { get; set; }

    public decimal Subtotal { get; set; }

    public decimal DiscountAmount { get; set; }

    public decimal TaxAmount { get; set; }

    public decimal TotalAmount { get; set; }

    public DateTime CreatedAtUtc { get; set; }

    public ICollection<InvoiceLine> Lines { get; set; }
        = new List<InvoiceLine>();
}